From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Wayne Chang <waynec@nvidia.com>
Date: Tue, 13 Aug 2024 02:40:11 +0000
Subject: [PATCH 056/241] NVIDIA: SAUCE: phy: tegra: Increase timeout for
 USB2_TRK_COMPLETED polling

The timeout for polling on USB2_TRK_COMPLETED in the UTMI bias pad
power-on sequence has been increased from 100 to 200 cycles.

This change is necessary due to the increased number of cycles required
to settle all biases in the new circuits. Additionally, the clock speed
of the tracking circuit has been reduced to 1 MHz, further contributing
to the need for a longer timeout.

http://nvbugs/4796326

Change-Id: I596954a5db1c4625da85b198de3ef2f50e2323b6
Signed-off-by: Wayne Chang <waynec@nvidia.com>
(cherry picked from commit de9a2c18f86248356c156f791243d7f264c1fdaa)
Reviewed-on: https://git-master.nvidia.com/r/c/3rdparty/canonical/linux-noble/+/3201120
Tested-by: Petlozu Pravareshwar <petlozup@nvidia.com>
Reviewed-by: Petlozu Pravareshwar <petlozup@nvidia.com>
GVS: buildbot_gerritrpt <buildbot_gerritrpt@nvidia.com>
Reviewed-by: Laxman Dewangan <ldewangan@nvidia.com>

diff --git a/drivers/phy/tegra/xusb-tegra186.c b/drivers/phy/tegra/xusb-tegra186.c
index 1d8a9e57ecf67..bf37b214776c9 100644
--- a/drivers/phy/tegra/xusb-tegra186.c
+++ b/drivers/phy/tegra/xusb-tegra186.c
@@ -638,7 +638,7 @@ static void tegra186_utmi_bias_pad_power_on(struct tegra_xusb_padctl *padctl)
 
 	if (padctl->soc->poll_trk_completed) {
 		err = padctl_readl_poll(padctl, XUSB_PADCTL_USB2_BIAS_PAD_CTL1,
-					USB2_TRK_COMPLETED, USB2_TRK_COMPLETED, 100);
+					USB2_TRK_COMPLETED, USB2_TRK_COMPLETED, 200);
 		if (err) {
 			/* The failure with polling on trk complete will not
 			 * cause the failure of powering on the bias pad.
-- 
2.43.0

